terraform/app/variables.tf

```diff
@@ -20,14 +21,27 @@ variable "account_id" {
   type        = string
   default     = "393416225559"
   description = "ID of aws account. Defaults to non-prod account."
-  nullable = false
+  nullable    = false
 }
 
-variable "domain_name" {
+variable "zone_name" {
   type        = string
   default     = "mavistesting.com"
   description = "Domain for which to create DNS certificate"
-  nullable = false
+  nullable    = false
+}
+
+variable "http_hosts" {
+  type = object({
+    MAVIS__HOST                        = string
+    MAVIS__GIVE_OR_REFUSE_CONSENT_HOST = string
+  })
+  description = "Https host names"
+  nullable    = true
+}
+
+locals {
+  host_headers = concat(tolist(toset(values(var.http_hosts))), [for v in tolist(toset(values(var.http_hosts))) : "www.${v}"])
```

Comment:
Is the second "tolist" really needed or could terraform iterate over the set?
Could you also extract `toset(values(var.http_hosts)` to a new locale, e.g. "unique_host_names"?